import org.gradle.api.tasks.testing.logging.TestExceptionFormat
import org.gradle.api.tasks.testing.logging.TestLogEvent

plugins {
    id 'org.jetbrains.kotlin.jvm'
}

description = 'mg-kot-util'

dependencies {

    // implementation 'mg.util:util:1.0-SNAPSHOT'

    // https://mvnrepository.com/artifact/commons-dbcp/commons-dbcp
    implementation group: 'commons-dbcp', name: 'commons-dbcp', version: '1.4'

    // https://mvnrepository.com/artifact/mysql/mysql-connector-java
    implementation group: 'mysql', name: 'mysql-connector-java', version: '5.1.17'

    // https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-core
    // implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.16.1'

    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-reflect', version: '2.0.0'

    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.3.1'
    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.3.1'
}

apply from: '../utils.gradle'

tasks.withType(Test) {

    useJUnitPlatform()

    def failedTests = rootProject.ext.failedTests
    afterTest { descriptor, result -> utils.appendFailed(descriptor, result, failedTests) }
    afterSuite { descriptor, result -> utils.formatSummary(descriptor, result, failedTests) }

    testLogging {
        events TestLogEvent.FAILED,
                TestLogEvent.PASSED,
                TestLogEvent.SKIPPED,
                TestLogEvent.STANDARD_ERROR,
                TestLogEvent.STANDARD_OUT

        exceptionFormat TestExceptionFormat.FULL
        showCauses true
        showExceptions true
        showStackTraces true
    }
}
